Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.swift

```diff
@@ -14,22 +14,26 @@
 
 import SwiftUI
 import ArcGIS
+import Network
 
 /// The `OfflineMapAreasView` component displays a list of downloadable preplanned map areas from a given web map.
 @MainActor
 @preconcurrency
 public struct OfflineMapAreasView: View {
     /// The view model for the map.
     @StateObject private var mapViewModel: MapViewModel
-    
+    /// The network monitor.
+    @StateObject private var networkMonitor = NetworkMonitor()
     /// The action to dismiss the view.
     @Environment(\.dismiss) private var dismiss: DismissAction
-    
     /// The web map to be taken offline.
     private let onlineMap: Map
-    
     /// The currently selected map.
     @Binding private var selectedMap: Map?
+    /// A Boolean value indicating whether the device has an internet connection.
+    @State private var isConnected = true
+    /// A Boolean value indicating whether the offline banner should be presented.
+    @State private var offlineBannerIsPresented = false
```

Comment:
I refactored the network connection logic in [191a3ff](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/pull/866/commits/191a3ff3d4a787febdb832033f0616c3ed0e69f6). `isConnected` should only be set when the view loads and upon pull to refresh while `offlineBannerIsPresented` should be updated upon network connection changes.

Reply:
Could `isConnected` be optional? Then `nil` could mean not determined yet. I'm just looking at three different properties related to connection (`connectionIsDetermined`, `isConnected`, and `offlineBannerIsPresented`) and can't help but wonder if they could be consolidated into two.

Reply:
While it may be good to get these 3 Booleans consolidated, upon further discussion today, Ryan suggested to not use the `NWPathMonitor` that proactively listen to network connectivity updates. Instead, we would use the loadStatus/error to determine whether the component has Internet access or not.

Thus, `offlineBannerIsPresented` would be gone shortly, and `connectionIsDetermined` might also be represented by the nullability of `isConnected` I guess

Reply:
p.s. I've discussed with Destiny. We can follow one of the 2 paths

- apply your suggestion here, and later refactor slightly to remove the `NWPathMonitor` related code
- branch off a separate PR (b) to remove `NWPathMonitor`, and resolve all other comments in this PR (a). First merge this PR (a), then merge the PR (b)

Reply:
Does that mean `NetworkMonitor` will be going away? If so, it doesn't make a whole lot of sense to merge it in this PR just to rip it out in a follow-up PR. Could it just be removed in this PR?

Reply:
Yes, it will go away. I was afraid that removing it would delay the review of this PR. But if you don't mind, Destiny can make the changes in this PR.

Reply:
Go for it!